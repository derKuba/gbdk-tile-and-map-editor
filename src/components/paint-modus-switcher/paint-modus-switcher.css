
.paint-checkbox {
  display: flex;
  margin-left: 20px;
}

.paint-checkbox label{
  margin-bottom: 10px;
  margin-right: 10px;
}

.paint-checkbox [type=radio] { 
    position: absolute;
    opacity: 0;
    width: 0;
    height: 0;
}
  
/* IMAGE STYLES */
.paint-checkbox [type=radio] + img {
  cursor: pointer;
}
  
/* CHECKED STYLES */
.paint-checkbox [type=radio]:checked + img {
  outline: 2px solid #33ccff;
}

.paint-checkbox .icon {
  width: 40px;
  height: 40px;
}