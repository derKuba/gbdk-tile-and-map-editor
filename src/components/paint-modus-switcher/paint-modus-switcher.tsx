import { Component, h, State, Prop } from '@stencil/core';

import { PaintModus } from '../../models/paintModus';

import Tunnel from '../../provider/StateProvider';

@Component({
    tag: 'mvjp-paint-modus-switcher',
    styleUrl: 'paint-modus-switcher.css',
    shadow: true,
})
export class PaintModusSwitcher {
    @State() paintModus: PaintModus = PaintModus.BRUSH;

    /**
     *   setter of paintmodus
     */
    @Prop() setPaintModus: Function;

    private changeModus = newModus => {
        this.paintModus = newModus;
        this.setPaintModus(newModus);
    };

    render() {
        return (
            <div class="paint-checkbox">
                <label>
                    <input
                        type="radio"
                        name="brush"
                        value={PaintModus.BRUSH}
                        checked={this.paintModus === PaintModus.BRUSH}
                        onClick={() => this.changeModus(PaintModus.BRUSH)}
                    />
                    <img class="icon" src={'./assets/icon/freepik_paintbrush.png'} alt="brush icon" />
                </label>

                <label>
                    <input
                        type="radio"
                        name="bucket"
                        value={PaintModus.BUCKET}
                        checked={this.paintModus === PaintModus.BUCKET}
                        onClick={() => this.changeModus(PaintModus.BUCKET)}
                    />
                    <img class="icon" src={'./assets/icon/freepik_paint_bucket.png'} alt="bucket icon" />
                </label>
            </div>
        );
    }
}

Tunnel.injectProps(PaintModusSwitcher, ['setPaintModus']); // eslint-disable-line @stencil/ban-side-effects
