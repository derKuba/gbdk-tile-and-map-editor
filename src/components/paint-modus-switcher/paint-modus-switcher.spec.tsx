import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { PaintModusSwitcher } from './paint-modus-switcher';

describe('mvjp-paint-modus-switcher', () => {
    it('builds', () => {
        expect(new PaintModusSwitcher()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [PaintModusSwitcher],
            template: () => {
                return <mvjp-paint-modus-switcher />;
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('should test navigate to', async () => {
        const setPaintModusMock = jest.fn();

        const page = await newSpecPage({
            components: [PaintModusSwitcher],
            template: () => {
                return <mvjp-paint-modus-switcher setPaintModus={setPaintModusMock} />;
            },
        });

        page.root.shadowRoot.querySelectorAll('input[name="bucket"]')[0].click();
        await page.waitForChanges();
        expect(setPaintModusMock).toBeCalledWith(1);

        page.root.shadowRoot.querySelectorAll('input[name="brush"]')[0].click();
    });
});
