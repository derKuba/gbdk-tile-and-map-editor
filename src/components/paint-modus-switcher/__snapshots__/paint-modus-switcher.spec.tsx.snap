// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-paint-modus-switcher matches snapshot 1`] = `
<mvjp-paint-modus-switcher>
  <mock:shadow-root>
    <div class="paint-checkbox">
      <label>
        <input checked="" name="brush" type="radio" value="0">
        <img alt="brush icon" class="icon" src="./assets/icon/freepik_paintbrush.png">
      </label>
      <label>
        <input name="bucket" type="radio" value="1">
        <img alt="bucket icon" class="icon" src="./assets/icon/freepik_paint_bucket.png">
      </label>
    </div>
  </mock:shadow-root>
</mvjp-paint-modus-switcher>
`;
